zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/SearchQueryRequestMapper.java

```diff
@@ -120,29 +131,185 @@ public static Either<ProblemDetail, ProcessInstanceSort> toSearchQuerySort(
+              ProblemDetail.forStatusAndDetail(
+                  HttpStatus.BAD_REQUEST, "Unknown sortOrder: " + order));
+        }
+      }
+
+      if (!validationErrors.isEmpty()) {
+        return Either.left(
+            validationErrors.get(0)); // Return the first encountered error for simplicity
+      }
+
+      return Either.right(builder.build());
+    }
+
+    return Either.right(null);
+  }
+
+  public static UserTaskFilter toUserTaskFilter(final UserTaskFilterRequest filter) {
+    final var builder = FilterBuilders.userTask();
+
+    if (filter != null) {
+      // key
+      if (filter.getKey() != null) {
+        builder.userTaskKeys(filter.getKey());
+      }
+
+      // state
+      if (filter.getTaskState() != null && !filter.getTaskState().isEmpty()) {
+        builder.userTaskState(filter.getTaskState());
+      }
+
+      // assignee
+      if (filter.getAssignee() != null && !filter.getAssignee().isEmpty()) {
+        builder.assignees(filter.getAssignee());
+      }
+
+      // candidateGroup
+      if (filter.getCandidateGroup() != null && !filter.getCandidateGroup().isEmpty()) {
+        builder.candidateGroups(filter.getCandidateGroup());
+      }
+
+      // candidateUser
+      if (filter.getCandidateUser() != null && !filter.getCandidateUser().isEmpty()) {
+        builder.candidateUsers(filter.getCandidateUser());
+      }
+
+      // processDefinitionKey
+      if (filter.getProcessDefinitionKey() != null) {
+        builder.processDefinitionKeys(filter.getProcessDefinitionKey());
+      }
+
+      // processInstanceKey
+      if (filter.getProcessInstanceKey() != null) {
+        builder.processInstanceKeys(filter.getProcessInstanceKey());
+      }
+
+      // tenantIds
+      if (filter.getTenantIds() != null) {
+        builder.tenantIds(filter.getTenantIds());
+      }
+
+      // creationTime
+      if (filter.getCreationTime() != null) {
+        builder.creationDate(toDateValueFilter(filter.getCreationTime()));
+      }
+
+      // completionTime
+      if (filter.getCompletionTime() != null) {
+        builder.completionDate(toDateValueFilter(filter.getCompletionTime()));
+      }
+
+      // dueDate
+      if (filter.getDueDate() != null) {
+        builder.dueDate(toDateValueFilter(filter.getDueDate()));
+      }
+
+      // followUpDate
+      if (filter.getFollowUpDate() != null) {
+        builder.followUpDate(toDateValueFilter(filter.getFollowUpDate()));
+      }
+    }
+
+    return builder.build();
+  }
+
+  public static Either<ProblemDetail, UserTaskQuery> toUserTaskQuery(
+      final UserTaskSearchQueryRequest request) {
+    final var pageResult = toSearchQueryPage(request.getPage());
+    if (pageResult.isLeft()) {
+      return Either.left(pageResult.getLeft());
+    }
+    final var page = pageResult.get();
+
+    final var sortingResult = toUserTaskSearchQuerySort(request.getSort());
+    if (sortingResult.isLeft()) {
+      return Either.left(sortingResult.getLeft());
+    }
```

Comment:
🔧 The collection of validation errors needs to go one step further. The page and sorting validation should return an `Either<List<String>, XYZ>` with `XYZ` being the success result of those methods respectively. In here, we can merge the String lists into one. If that lists is not empty, we can return an `Either.Left` with a `ProblemDetail`. Otherwise, we're again feeding only part of the validation errors back to the user.

Reply:
I have update this on the follow commits: 
- https://github.com/camunda/camunda/pull/19625/commits/a8c8a4b72fa4854b0cc70d1bec06dcddc4da2f05 
- https://github.com/camunda/camunda/pull/19625/commits/55bd74cf1ca948c641ad37081c3cf2675c85209f